src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,239 @@
+///<reference path="../../reference.ts" />
+
+module Plottable {
+export module Component {
+  export module Interactive {
+    type _EdgeIndicator = {
+      top: boolean;
+      bottom: boolean;
+      left: boolean;
+      right: boolean;
+    }
+
+    export class DragBoxLayer extends Component.SelectionBoxLayer {
+      private _dragInteraction: Interaction.Drag;
+      private _boxEdgeT: D3.Selection;
+      private _boxEdgeB: D3.Selection;
+      private _boxEdgeL: D3.Selection;
+      private _boxEdgeR: D3.Selection;
+      private _boxCornerTL: D3.Selection;
+      private _boxCornerTR: D3.Selection;
+      private _boxCornerBL: D3.Selection;
+      private _boxCornerBR: D3.Selection;
+      private _detectionRadius = 3;
+
+      private _xResizable = false;
+      private _yResizable = false;
+      private _grabbedEdges: _EdgeIndicator;
+      private _initTL: Point;
+      private _initBR: Point;
+
+      constructor() {
+        super();
+        this.classed("drag-box-layer", true);
+
+        this._dragInteraction = new Interaction.Drag();
+
+        this._dragInteraction.onDragStart((s: Point) => {
+          this.boxVisible(true);
+
+          this._grabbedEdges = this._getEdges(s);
+          if (!this._yResizable) {
+            this._grabbedEdges.top = false;
+            this._grabbedEdges.bottom = false;
+          }
+          if (!this._xResizable) {
+            this._grabbedEdges.left = false;
+            this._grabbedEdges.right = false;
+          }
+
+          if (!this._grabbedEdges.top && !this._grabbedEdges.bottom &&
+              !this._grabbedEdges.left && !this._grabbedEdges.right) {
+            this.bounds({
+              topLeft: s,
+              bottomRight: s
+            });
+            this._grabbedEdges.bottom = true;
+            this._grabbedEdges.right = true;
+          }
+          var bounds = this.bounds();
+          this._initTL = bounds.topLeft;
+          this._initBR = bounds.bottomRight;
+        });
+
+        this._dragInteraction.onDrag((s: Point, e: Point) => {
+          var topLeft = this._initTL;
+          var bottomRight = this._initBR;
+
+          if (this._grabbedEdges.bottom) {
+            bottomRight.y = e.y;
+          } else if (this._grabbedEdges.top) {
+            topLeft.y = e.y;
+          }
+
+          if (this._grabbedEdges.right) {
+            bottomRight.x = e.x;
+          } else if (this._grabbedEdges.left) {
+            topLeft.x = e.x;
+          }
+
+          this.bounds({
+            topLeft: topLeft,
+            bottomRight: bottomRight
+          });
+        });
+
+        this._dragInteraction.onDragEnd((s: Point, e: Point) => {
+          if (s.x === e.x && s.y === e.y) {
+            this.boxVisible(false);
+          }
```

Comment:
Would it be a good idea to have this.boxVisible(false) be tied to bounds with start/endPoint the same?  I'm mainly curious if this is a DragBox or a SelectionBox issue now.